DfciPkg/SettingsManager/SettingsManagerProvider.c

```diff
@@ -376,19 +372,23 @@ ProviderValueAsAscii (
         break;
       }
 
-      Value = AllocateZeroPool (ENABLED_STRING_SIZE);
+      if (v == ENABLE_INCONSISTENT) {
+        AsciiString = "Inconsistent";
+      } else if (v) {
+        AsciiString = "Enabled";
+      } else {
+        AsciiString = "Disabled";
+      }
+
+      ValueSize = AsciiStrLen (AsciiString) + 1;

```

Comment:
nit: should probably use SafeStrings, even though this is not user input. 
```suggestion
      ValueSize = AsciiStrnSizeS (AsciiString, DFCI_MAX_ID_LEN) + 1;

```

Reply:
Agree

Reply:
Also, the `Size()` functions (like `AsciiStrnSizeS()`) include the null terminator character.